labman/gui/handlers/process_handlers/test/test_base.py

```diff
@@ -0,0 +1,32 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2017-, labman development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from unittest import main
+
+from labman.gui.testing import TestHandlerBase
+from labman.gui.handlers.base import BaseDownloadHandler
+from labman.db.process import _Process
+
+
+class TestBaseDownloadHandler(TestHandlerBase):
```

Comment:
Would be nice to have tests for the deliver text and deliver zip methods.

Reply:
I am sad to admit that I spent *three hours* trying to write a working unit test for deliver_text directly and finally gave up ... the issue I encountered is that I can't use the same idiom used in all the other handler tests (that is, `response = self.get('/the/handler/url")` and then check the contents of the `response` object) because the BaseDownloadHandler isn't directly bound to any url (and I thought it seemed like overkill to bind it to one just for testing, but maybe not).  That means I had to fall back to a much more manual sort of testing as described at https://stackoverflow.com/questions/35560415/how-to-write-unit-tests-for-python-tornado-application#35562775 and I just couldn't get the mocking of various objects right ... I kept running up against errors because deep in the code somewhere the handler object was expecting to hold a response object that in fact didn't exist so methods called on it failed, or some such :(  If there are any working examples of a unit test like this somewhere in the Labman/Qiita/etc codebase, I'd be more than happy to copy off one!

Reply:
Sorry to hear that. In all fairness I didn't think about this enough to realize that this was an edge case. I think it is good to leave as is. This is indirectly exercised by other handlers, so at least we can catch things if they do go wrong.